packages/base-crypto/src/VirgilCrypto.ts

```diff
@@ -400,20 +606,29 @@ export class VirgilCrypto implements ICrypto {
     messageInfoCustomParams.addData(DATA_SIGNATURE_KEY, signature);
     messageInfoCustomParams.addData(DATA_SIGNER_ID_KEY, privateKey.identifier);
 
-    recipientCipher.startEncryption();
-    const messageInfo = recipientCipher.packMessageInfo();
-    const processEncryption = recipientCipher.processEncryption(myData);
-    const finishEncryption = recipientCipher.finishEncryption();
+    let encryptedData: BufferType | undefined;
+    let metadata: BufferType | undefined;
+    try {
+      recipientCipher.startEncryption();
+      const messageInfo = recipientCipher.packMessageInfo();
+      const processEncryption = recipientCipher.processEncryption(myData);
+      const finishEncryption = recipientCipher.finishEncryption();
+      encryptedData = NodeBuffer.concat([processEncryption, finishEncryption]);
+      metadata = toBuffer(messageInfo);
+    } catch (error) {
+      lowLevelPublicKeys.forEach(lowLevelPublicKey => lowLevelPublicKey.delete());
+      recipientCipher.delete();
+      aes256Gcm.delete();
+      messageInfoCustomParams.delete();
+      throw error;
+    }
 
+    lowLevelPublicKeys.forEach(lowLevelPublicKey => lowLevelPublicKey.delete());
     recipientCipher.delete();
     aes256Gcm.delete();
     messageInfoCustomParams.delete();
-    lowLevelPublicKeys.forEach(lowLevelPublicKey => lowLevelPublicKey.delete());
 
-    return {
-      encryptedData: NodeBuffer.concat([processEncryption, finishEncryption]),
-      metadata: toBuffer(messageInfo),
-    };
+    return { encryptedData, metadata };
```

Comment:
See previous comments about returning result from inside the `try..finally` block